lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3527,6 +3528,35 @@ void SPIRVToLLVM::transIntelFPGADecorations(SPIRVValue *BV, Value *V) {
   }
 }
 
+void SPIRVToLLVM::transUserSemantic(unsigned FunNum) {
+  auto Fun = BM->getFunction(FunNum);
+  auto transFun = transFunction(Fun);
+  for (auto UsSem : Fun->getDecorationStringLiteral(DecorationUserSemantic)) {
```

Comment:
Strictly speaking, having `UserSemantic` decoration on `Function` is against the spec, but we allow this for various purposes (like prototyping new features when we need to attach some information on function and propagate that through SPIR-V) - I suggest to leave a comment about this here in the code